spring-cloud-dataflow-configuration-metadata/src/main/java/org/springframework/cloud/dataflow/configuration/metadata/container/DefaultContainerImageMetadataResolver.java

```diff
@@ -48,12 +51,23 @@ public Map<String, String> getImageLabels(String imageName) {
 
 		ContainerRegistryRequest registryRequest = this.containerRegistryService.getRegistryRequest(imageName);
 
-		Map manifest = this.containerRegistryService.getImageManifest(registryRequest, Map.class);
-
-		if (manifest != null && !isNotNullMap(manifest.get("config"))) {
-			throw new ContainerRegistryException(
-					String.format("Image [%s] has incorrect or missing manifest config element: %s",
-							imageName, manifest.toString()));
+		Map<String, Object> manifest = this.containerRegistryService.getImageManifest(registryRequest, Map.class);
+
+		if (manifest != null && manifest.get("config") == null) {
+			// when both Docker and OCI images are stored in repository the response for OCI image when using Docker manifest type will not contain config.
+			// In the case where we don't receive a config and schemaVersion is less than 2 we try OCI manifest type.
+			String manifestMediaType = registryRequest.getRegistryConf().getManifestMediaType();
+			if (asInt(manifest.get("schemaVersion")) < 2
```

Comment:
Schema version maybe removed in the future so we need to handle a null condition.  https://github.com/opencontainers/image-spec/blob/main/manifest.md

Reply:
`asInt` will return 0 when `schemaVersion` is not present.

Reply:
For now this field is required.    But it says that it could be removed in the future.   So for now a null is invalid, but in the future a null may indicate something different.   So would this indicate an exception be thrown.

Reply:
If we cannot read the manifest we try something else. If that fails we throw an exception.